check_diff/src/lib.rs

```diff
@@ -0,0 +1,54 @@
+use std::env;
+use std::io::{self, Write};
+use std::path::Path;
+use std::process::Command;
+use tracing::{event, Level};
+
+pub struct GitErrors {
+    stdout: Vec<u8>,
+    stderr: Vec<u8>,
+}
+
+/// Clone a git repository
+///
+/// Parameters:
+/// url: git clone url
+/// dest: directory where the repo should be cloned
+pub fn clone_git_repo(url: &str, dest: &Path) -> Result<(), GitErrors> {
+    let git_cmd = Command::new("git")
+        .env("GIT_TERMINAL_PROMPT", "0")
+        .args([
+            "clone",
+            "--quiet",
+            url,
+            "--depth",
+            "1",
+            dest.to_str().unwrap(),
+        ])
+        .output()
+        .expect("failed to clone repository");
+    // if the git command does not return successfully,
+    // any command on the repo will fail. So fail fast.
+    if !git_cmd.status.success() {
+        io::stdout().write_all(&git_cmd.stdout).unwrap();
+        io::stderr().write_all(&git_cmd.stderr).unwrap();
+        let errors = GitErrors {
+            stderr: git_cmd.stderr,
+            stdout: git_cmd.stdout,
+        };
+        return Err(errors);
+    } else {
+        event!(Level::INFO, "Successfully clone repository.");
+        return Ok(());
+    }
+}
```

Comment:
If we're returning a `Result` I don't think we should call methods like `expect` or `unwrap` that could panic. Probably better to extend `GitError` with a `IO` variant.

Also, I don't think we should emit anything to `stdout` or `stderr` here. It's better not to emit anything from the library code and let the binary that we're builder handle all that.

Oh and you can simplify the `event!(..)` calls to just be `info!(..)`

```rust
use tracing::info;

enum GitError {
    FailedClone { stdout: Vec<u8>, stderr: Vec<u8> },
    IO(std::io::Error),
}

/// Clone a git repository
///
/// Parameters:
/// url: git clone url
/// dest: directory where the repo should be cloned
pub fn clone_git_repo(url: &str, dest: &Path) -> Result<(), GitError> {
    let git_cmd = Command::new("git")
        .env("GIT_TERMINAL_PROMPT", "0")
        .args([
            "clone",
            "--quiet",
            url,
            "--depth",
            "1",
            dest.to_str().unwrap(),
        ])
        .output()?;

    // if the git command does not return successfully,
    // any command on the repo will fail. So fail fast.
    if !git_cmd.status.success() {
        let error = GitError::CloneFailed { stdout: git_cmd.stdout, stderr: git_cmd.stderr };
        return Err(error);
    }

    info!("Successfully clone repository.");
    return Ok(());
}
```